src/object/cli_obj.c

```diff
@@ -2254,11 +2242,16 @@ obj_comp_cb(tse_task_t *task, void *data)
 		daos_gettime_coarse(&obj->cob_time_fetch_leader[idx]);
 	}
 
-	if (obj_auxi->map_ver_reply > obj_auxi->map_ver_req) {
-		D_DEBUG(DB_IO, "map_ver stale (req %d, reply %d).\n",
-			obj_auxi->map_ver_req, obj_auxi->map_ver_reply);
+	/* Check if the pool map needs to refresh */
+	if (obj_auxi->map_ver_reply > obj_auxi->map_ver_req ||
+	    daos_crt_network_error(task->dt_result) ||
+	    task->dt_result == -DER_STALE || task->dt_result == -DER_TIMEDOUT) {
```

Comment:
can we get any of the other failures like -DER_EVICTED here?

Reply:
Hmm, not sure if DER_EVICTED will be caused by stale pool map. Need check with others. I will update it if I need refresh the patch, otherwise I will add it in a new patch.